core/src/com/cloud/storage/template/HttpTemplateDownloader.java

```diff
@@ -117,7 +116,7 @@ public boolean retryMethod(final HttpMethod method, final IOException exception,
             request = new GetMethod(downloadUrl);
             request.getParams().setParameter(HttpMethodParams.RETRY_HANDLER, myretryhandler);
             completionCallback = callback;
-            //this.request.setFollowRedirects(false);
```

Comment:
In fact, this line has been commented since the very beginning! https://github.com/apache/cloudstack/commit/05c020e1f6679edb74ef9ee513fc6240db981139